### PR TITLE
better error message for to_pandas()

### PR DESCRIPTION
**Pull request by janerigby**
I've added a more helpful error message to to_pandas, that says which are the offending (multidimensionsal) columns that are making to_pandas() fail.

   @eteq

### CODE REVIEW

File: astropy/table/table.py [hunk 1]
```diff
@@ -2717,7 +2717,8 @@ def to_pandas(self):
             raise ValueError("Cannot convert a table with mixin columns to a pandas DataFrame")
 
         if any(getattr(col, 'ndim', 1) > 1 for col in self.columns.values()):
-            raise ValueError("Cannot convert a table with multi-dimensional columns to a pandas DataFrame")
+            badcols = [name for name,col in self.columns.items() if (getattr(col, 'ndim', 1) > 1)]
```
Comment (taldcroft) [inline astropy/table/table.py]: Move this out in front of the original `if` block e.g.:
```
badcols = [name for name,col in self.columns.items() if (getattr(col, 'ndim', 1) > 1)]
if badcols:
    ...
```
Also, please keep the line width < 90 characters and adhere to PEP8 (in particular space after `,`).

**Comment by bsipocz** [comment]
> For final approval this will also need an entry in the CHANGES.rst indicating this enhancement.

the type of the error hasn't changed, so this could be even backported. Do we need a changelog for the rephrased message?

**Comment by janerigby** [comment]
Tom and Brigitta,
This is my first ever pull request to astropy.  I don’t understand 40% of your emails, but I will try to google the relevant subjects and figure them out.  

FYI, what I submitted is an interim solution that makes the error message less cryptic.  I’m working on preventing an error, by teaching to_pandas() how to read multi-D columns, or skip them if it can’t.  Should I address your emails now, or keep working on the actual solution, and we’ll do the tests when I submit it?



Dr. Jane Rigby
Jane.Rigby@gmail.com
520-668-4290

**Comment by pllim** [comment]
@janerigby , the new error message does look more helpful, thanks!

@taldcroft , I have applied your changes but don't think it needs a change log since it only affects one error message. And also moved milestone forward since this is not a bug fix, but an enhancement. Thanks for the review.

**Comment by bsipocz** [comment]
Failure is unrelated, so I go ahead and merge this.

**Pull request merged**